tests/frontend/test_engine.py

```diff
@@ -146,13 +157,211 @@ def inspect():
+            "options", {"shots": inputs.shots}
+        )
+
+    def test_queue_job(self, starship_engine, monkeypatch):
+        mock_job = MagicMock()
+        monkeypatch.setattr("strawberryfields.engine.Job", mock_job)
+        mock_job_content = MagicMock()
+
+        result = starship_engine._queue_job(mock_job_content)
+        mock_job.assert_called_once_with(client=starship_engine.client)
+        result.manager.create.assert_called_once_with(circuit=mock_job_content)
+        assert starship_engine.jobs == [result]
+
+    def test__run_program(self, starship_engine, monkeypatch):
+        """
+        Tests StarshipEngine._run_program. Asserts that a program is converted to blackbird code,
+        compiled into a job content string and that the job is queued. Also asserts that a
+        completed job's result samples are returned.
+        """
+        mock_to_blackbird = MagicMock()
+        mock__get_blackbird = MagicMock()
+        program = MagicMock()
+        mock_job = MagicMock()
+        mock_job.is_complete = True
+        mock_job.is_failed = False
+
+        monkeypatch.setattr("strawberryfields.engine.to_blackbird", mock_to_blackbird)
+        monkeypatch.setattr(starship_engine, "_get_blackbird", mock__get_blackbird)
+        monkeypatch.setattr(starship_engine, "_queue_job", lambda job_content: mock_job)
+
+        some_params = {"param": MagicMock()}
+        result = starship_engine._run_program(program, **some_params)
+
+        mock__get_blackbird.assert_called_once_with(program=program, param=some_params["param"])
+
+        assert result == mock_job.result.result.value
+
+    def test__run_program_fails(self, starship_engine, monkeypatch):
+        """
+        Tests that an Exception is raised when a job has failed.
+        """
+        mock_to_blackbird = MagicMock()
+        mock__get_blackbird = MagicMock()
+        program = MagicMock()
+        mock_job = MagicMock()
+        mock_job.is_complete = False
+        mock_job.is_failed = True
+
+        monkeypatch.setattr("strawberryfields.engine.to_blackbird", mock_to_blackbird)
+        monkeypatch.setattr(starship_engine, "_get_blackbird", mock__get_blackbird)
+        monkeypatch.setattr(starship_engine, "_queue_job", lambda job_content: mock_job)
+
+        some_params = {"param": MagicMock()}
+
+        with pytest.raises(JobExecutionError):
+            starship_engine._run_program(program, **some_params)
+
+    def test__run(self, starship_engine, monkeypatch):
+        """
+        Tests StarshipEngine._run, with the assumption that the backend is a hardware backend
+        that supports running only a single program. This test ensures that a program is compiled
+        for the hardware backend, is locked, is added to self.run_progs, that it is run and that
+        a Result object is returned populated with the result samples.
+        """
+
+        inputs = MagicMock()
+        inputs.shots = 1
```

Comment:
would the test look any different if we had a value other than 1 here?

Reply:
It is expecting an integer. This was added to support a change in #111 -- we could use a `FuzzyInteger` which could be more appropriate.

Reply:
No plan to merge #111 immediately.

Probably enough to parameterize the test with two values, 1, and any other positive integer (say 1000)